src/main/java/org/folio/service/manager/input/InputDataManagerImpl.java

```diff
@@ -165,12 +168,49 @@ private void finalizeExport(ExportPayload exportPayload, ExportResult exportResu
     FileDefinition fileExportDefinition = exportPayload.getFileExportDefinition();
     String jobExecutionId = fileExportDefinition.getJobExecutionId();
     String tenantId = exportPayload.getOkapiConnectionParams().getTenantId();
-    jobExecutionService.updateJobStatusById(jobExecutionId, getJobExecutionStatus(exportResult), tenantId);
-    updateFileDefinitionStatusByResult(fileExportDefinition, exportResult, tenantId);
+    JobExecution.Status status = getJobExecutionStatus(exportResult);
+    if (status.equals(JobExecution.Status.COMPLETED)) {
+      isErrorsRelatedToUUIDsPresent(jobExecutionId, tenantId)
+          .onComplete(
+              isErrorsPresent -> {
+                if (isErrorsPresent.succeeded() && Boolean.TRUE.equals(isErrorsPresent.result())) {
+                  exportResult.setStatus(COMPLETED_WITH_ERRORS);
+                }
+                jobExecutionService.updateJobStatusById(
+                    jobExecutionId, getJobExecutionStatus(exportResult), tenantId);
+                updateFileDefinitionStatusByResult(fileExportDefinition, exportResult, tenantId);
+              });
+    } else {
+      jobExecutionService.updateJobStatusById(jobExecutionId, status, tenantId);
+      updateFileDefinitionStatusByResult(fileExportDefinition, exportResult, tenantId);
+    }
     closeSourceReader(jobExecutionId);
     removeInputDataContext(jobExecutionId);
   }
 
+  private Future<Boolean> isErrorsRelatedToUUIDsPresent(String jobExecutionId, String tenantId) {
+    Promise<Boolean> promise = Promise.promise();
+    List<Future> errorsList =
+        ErrorCode.reasonsAccordingToUUIDs().stream()
+            .map(
+                errorCode ->
+                    errorLogService.isErrorsByReasonPresent(errorCode, jobExecutionId, tenantId))
```

Comment:
so for each reason we are querying the DB, I think that will be a lot of DB calls, we can do a "IN" query instead, that way only 1 DB call is made

Reply:
updated with using `similar to` param